server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridParams.java

```diff
@@ -26,6 +26,7 @@
 final class GeoHashGridParams {
 
     /* recognized field names in JSON */
+    static final ParseField FIELD_TYPE = new ParseField("type");
```

Comment:
I think we should be more descriptive with the name here. Maybe we should use `hash_type`?

Reply:
We could rename GeoHashGridAggregationBuilder to GeoGridAggregationBuilder and then create a new GeoHashGridAggregationBuilder which is just a wrapper around GeoGridAggregationBuilder but hardcodes the type to provide backward compatibility. We would want to immediately deprecate the new GeoHashGridAggregationBuilder so we can remove it in 7.0.0. @jpountz do you think this is a good path to go down?

Reply:
:+1: